src/main/java/tatracker/model/module/Module.java

```diff
@@ -141,23 +112,33 @@ public boolean equals(Object other) {
         }
 
         Module otherModule = (Module) other;
-        return otherModule.getIdentifier().equals(this.getIdentifier());
+        return this.identifier.equals(otherModule.identifier);
     }
 
     @Override
     public int hashCode() {
-        // use this method for custom fields hashing instead of implementing your own
         return Objects.hash(identifier);
     }
 
     @Override
     public String toString() {
-        final StringBuilder builder = new StringBuilder();
-        builder.append(getName())
-                .append(" (")
-                .append(getIdentifier())
-                .append(") ")
-                .append(groupsString());
-        return builder.toString();
+        return String.format("%s (%s) %s", name, identifier, groupsString());
+    }
+
+    /**
+     * Returns a string that shows the value inside the groups list.
+     */
+    private String groupsString() {
```

Comment:
I think we no longer need groupsString() since the groups will be shown in UI. However, perhaps it is better to remove this method only once the UI has been completed?

Reply:
True! For now guess we could have that shown in the UI first. Then hopefully make it prettier!